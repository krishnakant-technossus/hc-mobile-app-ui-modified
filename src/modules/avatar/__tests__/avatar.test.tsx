// npx jest ./src/modules/avatar/__tests__/avatar.test.ts

import * as React from 'react';
import {fireEvent, render} from '@testing-library/react-native';
import {Text} from 'react-native';

import Avatar from '../avatar';
import ThemeProvider from '../../theming/theme-provider';
import defaultTheme from '../../theming/default-theme';

describe('Avatar', () => {
  describe('Rendering', () => {
    it('renders correctly', () => {
      const wrapper = render(
        <ThemeProvider theme={defaultTheme}>
          <Avatar uri="https://example.com/baby.jpg" />
        </ThemeProvider>,
      );

      const image = wrapper.getByTestId('avatar-image');

      expect(image).toBeDefined();

      expect(wrapper).toMatchSnapshot();
    });

    it('renders with the passed size', () => {
      const wrapper = render(
        <ThemeProvider theme={defaultTheme}>
          <Avatar size={240} uri="https://example.com/baby.jpg" />
        </ThemeProvider>,
      );

      const wrapperTile = render(
        <ThemeProvider theme={defaultTheme}>
          <Avatar size="tile" uri="https://example.com/baby.jpg" />
        </ThemeProvider>,
      );

      const wrapperProfile = render(
        <ThemeProvider theme={defaultTheme}>
          <Avatar size="profile" uri="https://example.com/baby.jpg" />
        </ThemeProvider>,
      );

      const image = wrapper.getByTestId('avatar-image');

      expect(image).toBeDefined();

      expect(wrapper).toMatchSnapshot();
      expect(wrapperTile).toMatchSnapshot();
      expect(wrapperProfile).toMatchSnapshot();
    });

    it('renders with the passed initials', () => {
      const wrapper = render(
        <ThemeProvider theme={defaultTheme}>
          <Avatar initials="AG" />
        </ThemeProvider>,
      );

      const initials = wrapper.getByText('AG');

      expect(initials).toBeDefined();

      expect(wrapper).toMatchSnapshot();
    });

    it('renders the passed marker correctly', () => {
      const wrapper = render(
        <ThemeProvider theme={defaultTheme}>
          <Avatar marker={<Text>O</Text>} uri="https://example.com/baby.jpg" />
        </ThemeProvider>,
      );

      const marker = wrapper.getByTestId('avatar-marker');

      expect(marker).toBeDefined();

      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('Interactions', () => {
    it('should call the onPress function on click', () => {
      const onPressMock = jest.fn();

      const {getByTestId} = render(
        <ThemeProvider theme={defaultTheme}>
          <Avatar uri="https://example.com/baby.jpg" onPress={onPressMock} />
        </ThemeProvider>,
      );

      const button = getByTestId('avatar-button');

      expect(button).toBeDefined();

      fireEvent.press(button);

      expect(onPressMock).toHaveBeenCalledTimes(1);
    });
  });
});
