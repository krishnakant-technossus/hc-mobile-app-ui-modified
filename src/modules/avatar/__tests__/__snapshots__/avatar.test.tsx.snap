// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar Rendering renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "height": 32,
      "justifyContent": "center",
      "opacity": 1,
      "overflow": "hidden",
      "width": 32,
    }
  }
  testID="avatar-button"
>
  <Image
    source={
      Object {
        "uri": "https://example.com/baby.jpg",
      }
    }
    style={
      Array [
        Object {
          "borderColor": "#FFFFFF",
          "borderRadius": 32,
          "borderWidth": 1,
          "height": "100%",
          "width": "100%",
        },
      ]
    }
    testID="avatar-image"
  />
</View>
`;

exports[`Avatar Rendering renders the passed marker correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "height": 32,
      "justifyContent": "center",
      "opacity": 1,
      "overflow": "hidden",
      "width": 32,
    }
  }
  testID="avatar-button"
>
  <Image
    source={
      Object {
        "uri": "https://example.com/baby.jpg",
      }
    }
    style={
      Array [
        Object {
          "borderColor": "#FFFFFF",
          "borderRadius": 32,
          "borderWidth": 1,
          "height": "100%",
          "width": "100%",
        },
      ]
    }
    testID="avatar-image"
  />
  <View
    style={
      Array [
        Object {
          "borderRadius": 12,
          "bottom": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
        },
      ]
    }
    testID="avatar-marker"
  >
    <Text>
      O
    </Text>
  </View>
</View>
`;

exports[`Avatar Rendering renders with the passed initials 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "height": 32,
      "justifyContent": "center",
      "opacity": 1,
      "overflow": "hidden",
      "width": 32,
    }
  }
  testID="avatar-button"
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#EDF1F5",
          "borderColor": "#DBE0E3",
          "borderRadius": 32,
          "borderWidth": 1,
          "height": "100%",
          "justifyContent": "center",
          "width": "100%",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#1C2530",
            "fontFamily": "System",
            "fontSize": 12,
            "fontStyle": "normal",
            "fontWeight": "400",
            "opacity": 0.75,
            "textAlign": "left",
          },
        ]
      }
      testID="text"
    >
      AG
    </Text>
  </View>
</View>
`;

exports[`Avatar Rendering renders with the passed size 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "height": 240,
      "justifyContent": "center",
      "opacity": 1,
      "overflow": "hidden",
      "width": 240,
    }
  }
  testID="avatar-button"
>
  <Image
    source={
      Object {
        "uri": "https://example.com/baby.jpg",
      }
    }
    style={
      Array [
        Object {
          "borderColor": "#FFFFFF",
          "borderRadius": 240,
          "borderWidth": 1,
          "height": "100%",
          "width": "100%",
        },
      ]
    }
    testID="avatar-image"
  />
</View>
`;

exports[`Avatar Rendering renders with the passed size 2`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "height": 36,
      "justifyContent": "center",
      "opacity": 1,
      "overflow": "hidden",
      "width": 36,
    }
  }
  testID="avatar-button"
>
  <Image
    source={
      Object {
        "uri": "https://example.com/baby.jpg",
      }
    }
    style={
      Array [
        Object {
          "borderColor": "#FFFFFF",
          "borderRadius": 36,
          "borderWidth": 1,
          "height": "100%",
          "width": "100%",
        },
      ]
    }
    testID="avatar-image"
  />
</View>
`;

exports[`Avatar Rendering renders with the passed size 3`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "height": 100,
      "justifyContent": "center",
      "opacity": 1,
      "overflow": "hidden",
      "width": 100,
    }
  }
  testID="avatar-button"
>
  <Image
    source={
      Object {
        "uri": "https://example.com/baby.jpg",
      }
    }
    style={
      Array [
        Object {
          "borderColor": "#FFFFFF",
          "borderRadius": 100,
          "borderWidth": 1,
          "height": "100%",
          "width": "100%",
        },
      ]
    }
    testID="avatar-image"
  />
</View>
`;
